-1

**Failed Tests**:   Build 
**Size**: This PR adds an extra 88KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-6e48cf/43801/summary.html
**COMMIT**: ea35f3d9fdef3ad78313e19c66025f82c84c414a
**CMSSW**: CMSSW_15_0_X_2025-01-16-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/47118/43801/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin src/GeneratorInterface/LHEInterface/plugins/LH5Source.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_15_0_X_2025-01-16-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_0_X_2025-01-16-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_0_X_2025-01-16-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/highfive/2.3.1-b005372960e0853c83418fdf7b27ed23/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/pcre/8.43-2d141998cfe5424b8f7aff48035cc2da/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/boost/1.80.0-24621bb5019ee7e7984682e0f80b5fcf/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/fastjet/3.4.1-43adababf6036a99696d8ed78b995b5c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/gsl/2.6-f7574c606b0ce57ff601d3ca9534cd01/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/hdf5/1.14.1-5dbe775e7ef35da13f2aee909660c0aa/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/hepmc/2.06.10-262d73524c32528e87adc31a98f68d52/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/lcg/root/6.32.09-2f25210d9b2d495c49b4a9094cb275ed/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/tbb/v2021.9.0-573155027234b8f945d29403a2749d52/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/zlib/1.2.13-d217cdbdd8d586e845e05946de2796be/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/fmt/10.2.1-e35fd1db5eb3abc8ac0452e8ee427196/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/hepmc3/3.2.7-f10dce5b16f0246f8cac8378933b9681/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/OpenBLAS/0.3.27-70a9dd2c9f309171934f13e3003b0540/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/openmpi/4.1.6-e8311880fc338dbd65c6aa8779ca36f5/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/tinyxml2/6.2.0-f99ae2781d074227d47e8a3e7c8ec87e/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/GeneratorInterface/LHEInterface/plugins/GeneratorInterfaceLHEIO/LH5Source.cc.d src/GeneratorInterface/LHEInterface/plugins/LH5Source.cc -o tmp/el8_amd64_gcc12/src/GeneratorInterface/LHEInterface/plugins/GeneratorInterfaceLHEIO/LH5Source.cc.o
>> Compiling edm plugin src/GeneratorInterface/LHEInterface/plugins/LHEProvenanceHelper.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_15_0_X_2025-01-16-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_15_0_X_2025-01-16-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_15_0_X_2025-01-16-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/highfive/2.3.1-b005372960e0853c83418fdf7b27ed23/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/pcre/8.43-2d141998cfe5424b8f7aff48035cc2da/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/boost/1.80.0-24621bb5019ee7e7984682e0f80b5fcf/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/fastjet/3.4.1-43adababf6036a99696d8ed78b995b5c/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/gsl/2.6-f7574c606b0ce57ff601d3ca9534cd01/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/hdf5/1.14.1-5dbe775e7ef35da13f2aee909660c0aa/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/hepmc/2.06.10-262d73524c32528e87adc31a98f68d52/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/lcg/root/6.32.09-2f25210d9b2d495c49b4a9094cb275ed/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/tbb/v2021.9.0-573155027234b8f945d29403a2749d52/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/xerces-c/3.1.3-c7b88eaa36d0408120f3c29826a04bf6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/zlib/1.2.13-d217cdbdd8d586e845e05946de2796be/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/fmt/10.2.1-e35fd1db5eb3abc8ac0452e8ee427196/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/hepmc3/3.2.7-f10dce5b16f0246f8cac8378933b9681/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/OpenBLAS/0.3.27-70a9dd2c9f309171934f13e3003b0540/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/openmpi/4.1.6-e8311880fc338dbd65c6aa8779ca36f5/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02872/el8_amd64_gcc12/external/tinyxml2/6.2.0-f99ae2781d074227d47e8a3e7c8ec87e/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v3 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/GeneratorInterface/LHEInterface/plugins/GeneratorInterfaceLHEIO/LHEProvenanceHelper.cc.d src/GeneratorInterface/LHEInterface/plugins/LHEProvenanceHelper.cc -o tmp/el8_amd64_gcc12/src/GeneratorInterface/LHEInterface/plugins/GeneratorInterfaceLHEIO/LHEProvenanceHelper.cc.o
src/GeneratorInterface/LHEInterface/plugins/LHEProvenanceHelper.cc: In constructor 'edm::LHEProvenanceHelper::LHEProvenanceHelper(const edm::TypeID&, const edm::TypeID&, edm::ProductRegistry&, edm::BranchIDListHelper&)':
src/GeneratorInterface/LHEInterface/plugins/LHEProvenanceHelper.cc:35:63: error: no matching function for call to 'edm::BranchDescription::BranchDescription(edm::BranchType, const char [7], const char [8], const char [16], const char [16], const char [1], const char [10], edm::ParameterSetID, edm::TypeWithDict, bool)'
   35 |                                                          false)),
      |                                                               ^
In file included from src/GeneratorInterface/LHEInterface/plugins/LHEProvenanceHelper.h:4,
                 from src/GeneratorInterface/LHEInterface/plugins/LHEProvenanceHelper.cc:3:
src/DataFormats/Provenance/interface/BranchDescription.h:51:5: note: candidate: 'edm::BranchDescription::BranchDescription(const edm::BranchDescription&, const std::string&, const std::string&)'
</pre>

--- a/Calibration/HcalAlCaRecoProducers/plugins/ProducerAnalyzer.cc
+++ b/Calibration/HcalAlCaRecoProducers/plugins/ProducerAnalyzer.cc
@@ -136,8 +136,8 @@ namespace cms {
     std::vector<StableProvenance const*> theProvenance;
     iEvent.getAllStableProvenance(theProvenance);
     for (auto const& provenance : theProvenance) {
-      edm::LogVerbatim("HcalAlCa") << " Print all module/label names " << provenance->moduleName() << " "
-                                   << provenance->moduleLabel() << " " << provenance->productInstanceName();
+      edm::LogVerbatim("HcalAlCa") << " Print all label names " << provenance->moduleLabel() << " "
+                                   << provenance->productInstanceName();
     }
 
     if (nameProd_ == "hoCalibProducer") {

--- a/DataFormats/Provenance/interface/BranchDescription.h
+++ b/DataFormats/Provenance/interface/BranchDescription.h
@@ -43,8 +43,6 @@ namespace edm {
                       std::string const& className,
                       std::string const& friendlyClassName,
                       std::string const& productInstanceName,
-                      std::string const& moduleName,
-                      ParameterSetID const& parameterSetID,
                       TypeWithDict const& theTypeWithDict,
                       bool produced = true,
                       bool availableOnlyAtEndTransition = false,
@@ -98,10 +96,6 @@ namespace edm {
     void setUnwrappedType(TypeWithDict const& type) { transient_.unwrappedType_ = type; }
     TypeID wrappedTypeID() const { return TypeID(transient_.wrappedType_.typeInfo()); }
     TypeID unwrappedTypeID() const { return TypeID(transient_.unwrappedType_.typeInfo()); }
-    int splitLevel() const { return transient_.splitLevel_; }
-    void setSplitLevel(int level) { transient_.splitLevel_ = level; }
-    int basketSize() const { return transient_.basketSize_; }
-    void setBasketSize(int size) { transient_.basketSize_ = size; }
 
     bool isSwitchAlias() const { return not transient_.switchAliasModuleLabel_.empty(); }
     std::string const& switchAliasModuleLabel() const { return transient_.switchAliasModuleLabel_; }
@@ -113,9 +107,6 @@ namespace edm {
 
     bool isProvenanceSetOnRead() const noexcept { return transient_.isProvenanceSetOnRead_; }
     void setIsProvenanceSetOnRead(bool value = true) noexcept { transient_.isProvenanceSetOnRead_ = value; }
-
-    ParameterSetID const& parameterSetID() const { return transient_.parameterSetID_; }
-    std::string const& moduleName() const { return transient_.moduleName_; }
 
     std::set<std::string> const& branchAliases() const { return branchAliases_; }
     void insertBranchAlias(std::string const& alias) { branchAliases_.insert(alias); }
@@ -136,14 +127,6 @@ namespace edm {
       Transients();
 
       void reset();
-
-      // The parameter set id of the producer.
-      // This is set if and only if produced_ is true.
-      ParameterSetID parameterSetID_;
-
-      // The module name of the producer.
-      // This is set if and only if produced_ is true.
-      std::string moduleName_;
 
       // The branch name, which is currently derivable from the other attributes.
       std::string branchName_;
@@ -166,14 +149,6 @@ namespace edm {
       // A TypeWithDict object for the unwrapped object
       // This is set if and only if the dropped_ is false
       TypeWithDict unwrappedType_;
-
-      // The split level of the branch, as marked
-      // in the data dictionary.
-      int splitLevel_;
-
-      // The basket size of the branch, as marked
-      // in the data dictionary.
-      int basketSize_;
 
       // Was this branch produced in this process rather than in a previous process
       bool produced_;

--- a/DataFormats/Provenance/interface/Provenance.h
+++ b/DataFormats/Provenance/interface/Provenance.h
@@ -53,7 +53,6 @@ namespace edm {
     std::string const& branchName() const { return stable().branchName(); }
     std::string const& className() const { return stable().className(); }
     std::string const& moduleLabel() const { return stable().moduleLabel(); }
-    std::string const& moduleName() const { return stable().moduleName(); }
     std::string const& processName() const { return stable().processName(); }
     std::string const& productInstanceName() const { return stable().productInstanceName(); }
     std::string const& friendlyClassName() const { return stable().friendlyClassName(); }

--- a/DataFormats/Provenance/interface/StableProvenance.h
+++ b/DataFormats/Provenance/interface/StableProvenance.h
@@ -41,7 +41,6 @@ namespace edm {
     std::string const& branchName() const { return branchDescription().branchName(); }
     std::string const& className() const { return branchDescription().className(); }
     std::string const& moduleLabel() const { return branchDescription().moduleLabel(); }
-    std::string const& moduleName() const { return branchDescription().moduleName(); }
     std::string const& processName() const { return branchDescription().processName(); }
     std::string const& productInstanceName() const { return branchDescription().productInstanceName(); }
     std::string const& friendlyClassName() const { return branchDescription().friendlyClassName(); }

--- a/DataFormats/Provenance/src/BranchDescription.cc
+++ b/DataFormats/Provenance/src/BranchDescription.cc
@@ -17,14 +17,10 @@ class TClass;
 
 namespace edm {
   BranchDescription::Transients::Transients()
-      : parameterSetID_(),
-        moduleName_(),
-        branchName_(),
+      : branchName_(),
         wrappedName_(),
         wrappedType_(),
         unwrappedType_(),
-        splitLevel_(),
-        basketSize_(),
         produced_(false),
         onDemand_(false),
         isTransform_(false),
@@ -55,8 +51,6 @@ namespace edm {
                                        std::string const& className,
                                        std::string const& friendlyClassName,
                                        std::string const& productInstanceName,
-                                       std::string const& moduleName,
-                                       ParameterSetID const& parameterSetID,
                                        TypeWithDict const& theTypeWithDict,
                                        bool produced,
                                        bool availableOnlyAtEndTransition,
@@ -73,8 +67,6 @@ namespace edm {
     setDropped(false);
     setProduced(produced);
     setOnDemand(false);
-    transient_.moduleName_ = moduleName;
-    transient_.parameterSetID_ = parameterSetID;
     transient_.availableOnlyAtEndTransition_ = availableOnlyAtEndTransition;
     setUnwrappedType(theTypeWithDict);
     init();
@@ -97,8 +89,6 @@ namespace edm {
     setProduced(aliasForBranch.produced());
     setOnDemand(false);  // will be re-set externally to the aliasForBranch.onDemand() after that one has been set
     transient_.availableOnlyAtEndTransition_ = aliasForBranch.availableOnlyAtEndTransition();
-    transient_.moduleName_ = aliasForBranch.moduleName();
-    transient_.parameterSetID_ = aliasForBranch.parameterSetID();
     setUnwrappedType(aliasForBranch.unwrappedType());
     init();
   }
@@ -164,8 +154,6 @@ namespace edm {
       // unwrapped type.
       setUnwrappedType(TypeWithDict::byName(fullClassName()));
       if (!bool(unwrappedType())) {
-        setSplitLevel(invalidSplitLevel);
-        setBasketSize(invalidBasketSize);
         setTransient(false);
         return;
       }
@@ -178,8 +166,6 @@ namespace edm {
     try {
       setWrappedType(wrType);
       if (!bool(wrappedType())) {
-        setSplitLevel(invalidSplitLevel);
-        setBasketSize(invalidBasketSize);
         return;
       }
     } catch (edm::Exception& caughtException) {
@@ -188,8 +174,6 @@ namespace edm {
     }
 
     setTransient(false);
-    setSplitLevel(invalidSplitLevel);
-    setBasketSize(invalidBasketSize);
     TDictAttributeMap* wp = wrappedType().getClass()->GetAttributeMap();
     if (wp && wp->HasKey("persistent") && !strcmp(wp->GetPropertyAsString("persistent"), "false")) {
       // Set transient if persistent == "false".
@@ -207,30 +191,10 @@ namespace edm {
             << "' contains an underscore ('_'), which is illegal in a module label.\n";
       }
     }
-
-    if (wp && wp->HasKey("splitLevel")) {
-      setSplitLevel(strtol(wp->GetPropertyAsString("splitLevel"), nullptr, 0));
-      if (splitLevel() < 0) {
-        throw cms::Exception("IllegalSplitLevel") << "' An illegal ROOT split level of " << splitLevel()
-                                                  << " is specified for class " << wrappedName() << ".'\n";
-      }
-      setSplitLevel(splitLevel() + 1);  //Compensate for wrapper
-    }
-    if (wp && wp->HasKey("basketSize")) {
-      setBasketSize(strtol(wp->GetPropertyAsString("basketSize"), nullptr, 0));
-      if (basketSize() <= 0) {
-        throw cms::Exception("IllegalBasketSize") << "' An illegal ROOT basket size of " << basketSize()
-                                                  << " is specified for class " << wrappedName() << "'.\n";
-      }
-    }
   }
 
   void BranchDescription::merge(BranchDescription const& other) {
     branchAliases_.insert(other.branchAliases().begin(), other.branchAliases().end());
-    if (splitLevel() == invalidSplitLevel)
-      setSplitLevel(other.splitLevel());
-    if (basketSize() == invalidBasketSize)
-      setBasketSize(other.basketSize());
   }
 
   void BranchDescription::setSwitchAliasForBranch(BranchDescription const& aliasForBranch) {
@@ -287,9 +251,6 @@ namespace edm {
 
     if (friendlyClassName_.empty())
       throwExceptionWithText("Friendly class name is not allowed to be empty");
-
-    if (produced() && !parameterSetID().isValid())
-      throwExceptionWithText("Invalid ParameterSetID detected");
   }
 
   void BranchDescription::updateFriendlyClassName() {

--- a/FWCore/Framework/src/ProductRegistryHelper.cc
+++ b/FWCore/Framework/src/ProductRegistryHelper.cc
@@ -81,8 +81,6 @@ namespace edm {
                               p->typeID_.userClassName(),
                               p->typeID_.friendlyClassName(),
                               p->productInstanceName_,
-                              iDesc.moduleName(),
-                              iDesc.parameterSetID(),
                               type,
                               true,
                               isEndTransition(p->transition_));

--- a/FWCore/Framework/test/Event_t.cpp
+++ b/FWCore/Framework/test/Event_t.cpp
@@ -222,8 +222,6 @@ void testEvent::registerProduct(std::string const& tag,
                            product_type.userClassName(),
                            product_type.friendlyClassName(),
                            productInstanceName,
-                           moduleClassName,
-                           moduleParams.id(),
                            product_type);
 
   moduleDescriptions_[tag] = ModuleDescription(
@@ -371,8 +369,6 @@ testEvent::testEvent()
                            product_type.userClassName(),
                            product_type.friendlyClassName(),
                            productInstanceName,
-                           moduleClassName,
-                           moduleParams.id(),
                            product_type);
 
   availableProducts_->addProduct(branch);

--- a/FWCore/Framework/test/ProductSelector_t.cpp
+++ b/FWCore/Framework/test/ProductSelector_t.cpp
@@ -43,9 +43,6 @@ TEST_CASE("test ProductSelector", "[ProductSelector]") {
   auto processConfiguration = std::make_shared<edm::ProcessConfiguration>();
   processConfiguration->setParameterSetID(dummyProcessPset.id());
 
-  edm::ParameterSet pset;
-  pset.registerIt();
-
   edm::TypeWithDict dummyTypeWithDict;
   // We pretend to have one module, with two products. The products
   // are of the same and, type differ in instance name.
@@ -56,10 +53,8 @@ TEST_CASE("test ProductSelector", "[ProductSelector]") {
   modAparams.registerIt();
   psetsA.insert(modAparams.id());
 
-  edm::BranchDescription b1(
-      edm::InEvent, "modA", "PROD", "UglyProdTypeA", "ProdTypeA", "i1", "", pset.id(), dummyTypeWithDict);
-  edm::BranchDescription b2(
-      edm::InEvent, "modA", "PROD", "UglyProdTypeA", "ProdTypeA", "i2", "", pset.id(), dummyTypeWithDict);
+  edm::BranchDescription b1(edm::InEvent, "modA", "PROD", "UglyProdTypeA", "ProdTypeA", "i1", dummyTypeWithDict);
+  edm::BranchDescription b2(edm::InEvent, "modA", "PROD", "UglyProdTypeA", "ProdTypeA", "i2", dummyTypeWithDict);
 
   // Our second pretend module has only one product, and gives it no
   // instance name.
@@ -69,16 +64,13 @@ TEST_CASE("test ProductSelector", "[ProductSelector]") {
   modBparams.registerIt();
   psetsB.insert(modBparams.id());
 
-  edm::BranchDescription b3(
-      edm::InEvent, "modB", "HLT", "UglyProdTypeB", "ProdTypeB", "", "", pset.id(), dummyTypeWithDict);
+  edm::BranchDescription b3(edm::InEvent, "modB", "HLT", "UglyProdTypeB", "ProdTypeB", "", dummyTypeWithDict);
 
   // Our third pretend is like modA, except it hass processName_ of
   // "USER"
 
-  edm::BranchDescription b4(
-      edm::InEvent, "modA", "USER", "UglyProdTypeA", "ProdTypeA", "i1", "", pset.id(), dummyTypeWithDict);
-  edm::BranchDescription b5(
-      edm::InEvent, "modA", "USER", "UglyProdTypeA", "ProdTypeA", "i2", "", pset.id(), dummyTypeWithDict);
+  edm::BranchDescription b4(edm::InEvent, "modA", "USER", "UglyProdTypeA", "ProdTypeA", "i1", dummyTypeWithDict);
+  edm::BranchDescription b5(edm::InEvent, "modA", "USER", "UglyProdTypeA", "ProdTypeA", "i2", dummyTypeWithDict);
 
   // These are pointers to all the branches that are available. In a
   // framework program, these would come from the ProductRegistry

--- a/FWCore/Framework/test/event_getrefbeforeput_t.cppunit.cc
+++ b/FWCore/Framework/test/event_getrefbeforeput_t.cppunit.cc
@@ -150,18 +150,8 @@ void testEventGetRefBeforePut::getRefTest() {
   auto processConfiguration = std::make_shared<edm::ProcessConfiguration>();
   processConfiguration->setParameterSetID(dummyProcessPset.id());
 
-  edm::ParameterSet pset;
-  pset.registerIt();
-
-  edm::BranchDescription product(edm::InEvent,
-                                 label,
-                                 processName,
-                                 dummytype.userClassName(),
-                                 className,
-                                 productInstanceName,
-                                 "",
-                                 pset.id(),
-                                 dummytype);
+  edm::BranchDescription product(
+      edm::InEvent, label, processName, dummytype.userClassName(), className, productInstanceName, dummytype);
 
   product.init();
 

--- a/FWCore/Framework/test/eventprincipal_t.cppunit.cc
+++ b/FWCore/Framework/test/eventprincipal_t.cppunit.cc
@@ -122,8 +122,6 @@ std::shared_ptr<edm::BranchDescription> test_ep::fake_single_process_branch(std:
                                                          productClassName,
                                                          friendlyProductClassName,
                                                          productInstanceName,
-                                                         moduleClass,
-                                                         modParams.id(),
                                                          dummyType);
   branchDescriptions_[tag] = result;
   return result;

--- a/FWCore/Framework/test/generichandle_t.cppunit.cc
+++ b/FWCore/Framework/test/generichandle_t.cppunit.cc
@@ -145,15 +145,8 @@ void testGenericHandle::getbyLabelTest() {
   edm::ParameterSet pset;
   pset.registerIt();
 
-  edm::BranchDescription product(edm::InEvent,
-                                 label,
-                                 processName,
-                                 dummytype.userClassName(),
-                                 className,
-                                 productInstanceName,
-                                 "",
-                                 pset.id(),
-                                 dummytype);
+  edm::BranchDescription product(
+      edm::InEvent, label, processName, dummytype.userClassName(), className, productInstanceName, dummytype);
 
   product.init();
 

--- a/FWCore/Framework/test/productregistry.cppunit.cc
+++ b/FWCore/Framework/test/productregistry.cppunit.cc
@@ -83,8 +83,6 @@ namespace {
                                   iDesc.fullClassName(),
                                   iDesc.friendlyClassName(),
                                   iDesc.productInstanceName() + "-" + name_,
-                                  "",
-                                  iDesc.parameterSetID(),
                                   iDesc.unwrappedType());
       reg_->addProduct(prod);
     }
@@ -102,20 +100,13 @@ void testProductRegistry::setUp() {
   edm::ParameterSet pset;
   pset.registerIt();
   intBranch_ = std::make_shared<edm::BranchDescription>(
-      edm::InEvent, "labeli", "PROD", "int", "int", "int", "", pset.id(), edm::TypeWithDict(typeid(int)));
+      edm::InEvent, "labeli", "PROD", "int", "int", "int", edm::TypeWithDict(typeid(int)));
 
   floatBranch_ = std::make_shared<edm::BranchDescription>(
-      edm::InEvent, "labelf", "PROD", "float", "float", "float", "", pset.id(), edm::TypeWithDict(typeid(float)));
+      edm::InEvent, "labelf", "PROD", "float", "float", "float", edm::TypeWithDict(typeid(float)));
 
-  intVecBranch_ = std::make_shared<edm::BranchDescription>(edm::InEvent,
-                                                           "labelvi",
-                                                           "PROD",
-                                                           "std::vector<int>",
-                                                           "ints",
-                                                           "vint",
-                                                           "",
-                                                           pset.id(),
-                                                           edm::TypeWithDict(typeid(std::vector<int>)));
+  intVecBranch_ = std::make_shared<edm::BranchDescription>(
+      edm::InEvent, "labelvi", "PROD", "std::vector<int>", "ints", "vint", edm::TypeWithDict(typeid(std::vector<int>)));
 
   simpleVecBranch_ =
       std::make_shared<edm::BranchDescription>(edm::InEvent,
@@ -124,8 +115,6 @@ void testProductRegistry::setUp() {
                                                "edm::OwnVector<edmtest::Simple>",
                                                "edmtestSimplesOwned",
                                                "ovsimple",
-                                               "",
-                                               pset.id(),
                                                edm::TypeWithDict(typeid(edm::OwnVector<edmtest::Simple>)));
   simpleDerivedVecBranch_ =
       std::make_shared<edm::BranchDescription>(edm::InEvent,
@@ -134,8 +123,6 @@ void testProductRegistry::setUp() {
                                                "edm::OwnVector<edmtest::SimpleDerived>",
                                                "edmtestSimpleDerivedsOwned",
                                                "ovsimplederived",
-                                               "",
-                                               pset.id(),
                                                edm::TypeWithDict(typeid(edm::OwnVector<edmtest::SimpleDerived>)));
 }
 

--- a/FWCore/Framework/test/test_catch2notTP_MergeableRunProductMetadata.cc
+++ b/FWCore/Framework/test/test_catch2notTP_MergeableRunProductMetadata.cc
@@ -65,8 +65,6 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
 
   SECTION("test main functions") {
     edm::ProductRegistry productRegistry;
-    edm::ParameterSet dummyPset;
-    dummyPset.registerIt();
 
     // not mergeable
     edm::BranchDescription prod1(edm::InRun,
@@ -75,8 +73,6 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
                                  "edmtest::Thing",
                                  "edmtestThing",
                                  "instance",
-                                 "aModule",
-                                 dummyPset.id(),
                                  edm::TypeWithDict::byName("edmtest::Thing"),
                                  false);
     productRegistry.copyProduct(prod1);
@@ -88,8 +84,6 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
                                  "edmtest::ThingWithMerge",
                                  "edmtestThingWithMerge",
                                  "instance",
-                                 "aModule",
-                                 dummyPset.id(),
                                  edm::TypeWithDict::byName("edmtest::ThingWithMerge"),
                                  false);
     productRegistry.copyProduct(prod2);
@@ -101,8 +95,6 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
                                  "edmtest::ThingWithMerge",
                                  "edmtestThingWithMerge",
                                  "instance",
-                                 "aModule",
-                                 dummyPset.id(),
                                  edm::TypeWithDict::byName("edmtest::ThingWithMerge"),
                                  false);
     productRegistry.copyProduct(prod3);
@@ -114,8 +106,6 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
                                  "edmtest::ThingWithMerge",
                                  "edmtestThingWithMerge",
                                  "instance",
-                                 "aModule",
-                                 dummyPset.id(),
                                  edm::TypeWithDict::byName("edmtest::ThingWithMerge"),
                                  true);
     productRegistry.addProduct(prod4);
@@ -127,8 +117,6 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
                                  "edmtest::ThingWithMerge",
                                  "edmtestThingWithMerge",
                                  "instance",
-                                 "aModule",
-                                 dummyPset.id(),
                                  edm::TypeWithDict::byName("edmtest::ThingWithMerge"),
                                  false);
     prod5.setDropped(true);
@@ -141,8 +129,6 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
                                  "edmtest::ThingWithMerge",
                                  "edmtestThingWithMerge",
                                  "instance",
-                                 "aModule",
-                                 dummyPset.id(),
                                  edm::TypeWithDict::byName("edmtest::ThingWithMerge"),
                                  false);
     productRegistry.copyProduct(prod6);
@@ -154,8 +140,6 @@ TEST_CASE("test MergeableRunProductMetadata", "[MergeableRunProductMetadata]") {
                                  "edmtest::ThingWithMerge",
                                  "edmtestThingWithMerge",
                                  "instance",
-                                 "aModule",
-                                 dummyPset.id(),
                                  edm::TypeWithDict::byName("edmtest::ThingWithMerge"),
                                  false);
     productRegistry.copyProduct(prod7);

--- a/FWCore/Integration/plugins/DelayedReaderThrowingSource.cc
+++ b/FWCore/Integration/plugins/DelayedReaderThrowingSource.cc
@@ -97,8 +97,6 @@ namespace edm {
                                            twd.userClassName(),
                                            twd.friendlyClassName(),
                                            "",  //product instance name
-                                           "",  //module name which isn't set for items not produced
-                                           dummy.id(),
                                            twd,
                                            false  //not produced
                                            ));

--- a/FWCore/Integration/plugins/PutOrMergeTestSource.cc
+++ b/FWCore/Integration/plugins/PutOrMergeTestSource.cc
@@ -38,7 +38,6 @@ namespace edmtest {
     void readEvent_(EventPrincipal& eventPrincipal) final;
 
     int stage_;
-    ParameterSet const dummyPSet_;
     BranchDescription thingDesc_;
     BranchDescription thingWithMergeDesc_;
     BranchDescription thingWithEqualDesc_;
@@ -51,19 +50,12 @@ using namespace edmtest;
 PutOrMergeTestSource::PutOrMergeTestSource(ParameterSet const& iPS, InputSourceDescription const& iISD)
     : InputSource(iPS, iISD),
       stage_(0),
-      dummyPSet_([]() {
-        ParameterSet dummy;
-        dummy.registerIt();
-        return dummy;
-      }()),
       thingDesc_(InRun,
                  "thingWithMergeProducer",
                  "PROD",
                  "edmtest::Thing",
                  "edmtestThing",
                  "endRun",
-                 "PutOrMergeTestSource",
-                 dummyPSet_.id(),
                  edm::TypeWithDict(typeid(edmtest::Thing)),
                  false,
                  true),
@@ -73,8 +65,6 @@ PutOrMergeTestSource::PutOrMergeTestSource(ParameterSet const& iPS, InputSourceD
                           "edmtest::ThingWithMerge",
                           "edmtestThingWithMerge",
                           "endRun",
-                          "PutOrMergeTestSource",
-                          dummyPSet_.id(),
                           edm::TypeWithDict(typeid(edmtest::ThingWithMerge)),
                           false,
                           true),
@@ -84,8 +74,6 @@ PutOrMergeTestSource::PutOrMergeTestSource(ParameterSet const& iPS, InputSourceD
                           "edmtest::ThingWithIsEqual",
                           "edmtestThingWithIsEqual",
                           "endRun",
-                          "PutOrMergeTestSource",
-                          dummyPSet_.id(),
                           edm::TypeWithDict(typeid(edmtest::ThingWithIsEqual)),
                           false,
                           true) {

--- a/FWCore/Sources/src/DaqProvenanceHelper.cc
+++ b/FWCore/Sources/src/DaqProvenanceHelper.cc
@@ -28,8 +28,6 @@ namespace {
                                 collectionName,
                                 friendlyName,
                                 "",
-                                sourceLabel,
-                                edm::ParameterSetID(),
                                 edm::TypeWithDict(rawDataType.typeInfo()),
                                 false);
     desc.setIsProvenanceSetOnRead();
@@ -55,7 +53,6 @@ namespace edm {
     // We don't currently use the untracked parameters, However, we make them available, just in case.
     std::string const& moduleLabel = constBranchDescription_.moduleLabel();
     std::string const& processName = constBranchDescription_.processName();
-    std::string const& moduleName = constBranchDescription_.moduleName();
     typedef std::vector<std::string> vstring;
     vstring empty;
 
@@ -71,7 +68,7 @@ namespace edm {
     ParameterSet pseudoInput;
     pseudoInput.addParameter<std::string>("@module_edm_type", "Source");
     pseudoInput.addParameter<std::string>("@module_label", moduleLabel);
-    pseudoInput.addParameter<std::string>("@module_type", moduleName);
+    pseudoInput.addParameter<std::string>("@module_type", sourceLabel);
     processParameterSet_.addParameter<ParameterSet>(moduleLabel, pseudoInput);
 
     processParameterSet_.addParameter<vstring>("@all_esmodules", empty);

--- a/FWCore/TestProcessor/src/TestProcessor.cc
+++ b/FWCore/TestProcessor/src/TestProcessor.cc
@@ -137,8 +137,6 @@ namespace edm {
                                        twd.userClassName(),
                                        twd.friendlyClassName(),
                                        produce.instanceLabel_,
-                                       "",
-                                       psetid,
                                        twd,
                                        true  //force this to come from 'source'
         );

--- a/GeneratorInterface/LHEInterface/plugins/LHEProvenanceHelper.cc
+++ b/GeneratorInterface/LHEInterface/plugins/LHEProvenanceHelper.cc
@@ -29,8 +29,6 @@ namespace edm {
                                                          "LHEEventProduct",
                                                          "LHEEventProduct",
                                                          "",
-                                                         "LHESource",
-                                                         ParameterSetID(),
                                                          TypeWithDict(eventProductType.typeInfo()),
                                                          false)),
         runProductBranchDescription_(BranchDescription(InRun,
@@ -41,8 +39,6 @@ namespace edm {
                                                        "LHERunInfoProduct",
                                                        "LHERunInfoProduct",
                                                        "",
-                                                       "LHESource",
-                                                       ParameterSetID(),
                                                        TypeWithDict(runProductType.typeInfo()),
                                                        false)),
         eventProductProvenance_(eventProductBranchDescription_.branchID()),
@@ -67,7 +63,7 @@ namespace edm {
     ParameterSet pset;
     std::string const& moduleLabel = eventProductBranchDescription_.moduleLabel();
     std::string const& processName = eventProductBranchDescription_.processName();
-    std::string const& moduleName = eventProductBranchDescription_.moduleName();
+    std::string const moduleName = "LHESource";
     typedef std::vector<std::string> vstring;
     vstring empty;
 

--- a/IOPool/Output/src/PoolOutputModule.cc
+++ b/IOPool/Output/src/PoolOutputModule.cc
@@ -26,6 +26,7 @@
 #include "TBranchElement.h"
 #include "TObjArray.h"
 #include "RVersion.h"
+#include "TDictAttributeMap.h"
 
 #include <fstream>
 #include <iomanip>
@@ -211,13 +212,31 @@ namespace edm {
         splitLevel = theBranch->GetSplitLevel();
         basketSize = theBranch->GetBasketSize();
       } else {
-        splitLevel = (prod.splitLevel() == BranchDescription::invalidSplitLevel ? splitLevel_ : prod.splitLevel());
+        auto wp = prod.wrappedType().getClass()->GetAttributeMap();
+        auto wpSplitLevel = BranchDescription::invalidSplitLevel;
+        if (wp && wp->HasKey("splitLevel")) {
+          wpSplitLevel = strtol(wp->GetPropertyAsString("splitLevel"), nullptr, 0);
+          if (wpSplitLevel < 0) {
+            throw cms::Exception("IllegalSplitLevel") << "' An illegal ROOT split level of " << wpSplitLevel
+                                                      << " is specified for class " << prod.wrappedName() << ".'\n";
+          }
+          wpSplitLevel += 1;  //Compensate for wrapper
+        }
+        splitLevel = (wpSplitLevel == BranchDescription::invalidSplitLevel ? splitLevel_ : wpSplitLevel);
         for (auto const& b : specialSplitLevelForBranches_) {
           if (b.match(prod.branchName())) {
             splitLevel = b.splitLevel_;
           }
         }
-        basketSize = (prod.basketSize() == BranchDescription::invalidBasketSize ? basketSize_ : prod.basketSize());
+        auto wpBasketSize = BranchDescription::invalidBasketSize;
+        if (wp && wp->HasKey("basketSize")) {
+          wpBasketSize = strtol(wp->GetPropertyAsString("basketSize"), nullptr, 0);
+          if (wpBasketSize <= 0) {
+            throw cms::Exception("IllegalBasketSize") << "' An illegal ROOT basket size of " << wpBasketSize
+                                                      << " is specified for class " << prod.wrappedName() << "'.\n";
+          }
+        }
+        basketSize = (wpBasketSize == BranchDescription::invalidBasketSize ? basketSize_ : wpBasketSize);
       }
       outputItemList.emplace_back(&prod, kept.second, splitLevel, basketSize);
     }

